Source/WebCore/page/ScreenOrientation.cpp

```diff
@@ -118,7 +118,7 @@ void ScreenOrientation::lock(LockType lockType, Ref<DeferredPromise>&& promise)
 
     if (document->settings().fullscreenRequirementForScreenOrientationLockingEnabled()) {
 #if ENABLE(FULLSCREEN_API)
-        if (!document->fullscreenManager().isFullscreen()) {
+        if (!document->fullscreenManager().fullscreenElement()) {
```

Comment:
This method more correct according to @nt1m. We should really rename `isFullscreen()` to something else to avoid confusion.

Reply:
It is not more correct. They are identical but I think the old name is clearer.

Reply:
@cdumez They are not identical when content inside an iframe is fullscreened, isFullscreen() will return false on the ancestor document, while fullscreenElement() will return the iframe element being implicitly fullscreened.

isFullscreen() and currentFullscreenElement() are identical however. It’s been this way for quite a long time, but I think it can probably be cleaned up.

Reply:
Oh I see, `fullscreenElement()` doesn't actually return `m_fullscreenElement`. A little surprising but OK. Seems this code could use some refactoring / renaming for clarity.

Reply:
Either way, it is a good idea to make this change separately, with a test.

Reply:
will do.